modules/ROOT/pages/_partials/sync/server-data-access.adoc

```diff
@@ -1,3 +1,125 @@
 = Data Access
 
-TODO
\ No newline at end of file
+It is very likely a {sync-service} application will need to access data from a database for it to work. The GraphQL spec or Voyager libraries does not require an application to use a particular type of database, and the developer of the application can choose a database that is most suitable for the application. 
+
+From a very high level, here are the steps to access data from a database in a {sync-service} application:
+
+. Initialize the database connection on app start
+. Add the database connection to the `context` object
+. Retrieve/persist data to the database in the application's resolvers
+
+== Initialize Database Connection
+
+Generally speaking, the application should set up the database connection on start, and fail if the connection can not be established. In order to connect to the database, usually the information like the hostname and port number of the database server, the credentials to authenticate etc will be required. It is recommended to use environment variables to set these details. 
+
+If the application is going to be deployed on OpenShift as described in the <<Running {sync-service} on OpenShift>> section, then the following environment variables will be setup for the application automatically:
+
+[options="header"]
+|====
+|Field Name|Environment Variable Name
+|Database Server Hostname|DB_HOSTNAME
+|Database Server Port|DB_PORT
+|Database Server Username|DB_USERNAME
+|Database Server Password|DB_PASSWORD
+|Database name|DB_NAME
+|Use SSL|DB_SSL
+|====
+
+It is highly recommended that the application use these environment variables to setup the database connection.
+
+For example, an application needs to connect to a PostgreSQL database. The database, table and schema is already setup. On app start, the database connection can be established like this:
+
+[source,javascript]
+----
+//knex is a SQL query builder
+const knex = require('knex')
+
+const dbConfig = {
+  database: process.env.DB_NAME || 'users',
+  user: process.env.DB_USERNAME || 'postgresql',
+  password: process.env.DB_PASSWORD || 'postgres',
+  host: process.env.DB_HOSTNAME || '127.0.0.1',
+  port: process.env.DB_PORT || '5432'
+}
+
+const db = knex({
+  client: 'pg',
+  connection: dbConfig
+})
+----
+
+== Add Database Connection to `context` object
+
+Once the database connection is established, it needs to be accessible from a resolver so that it can be used to retrieve or persist data. The recommended way to achieve this is to add the database connection to the `context` object. 
```

Comment:
kinda think this para is unnecessary

Reply:
why?

Reply:
or maybe reword, something like:

Add the database connection to the `context` object so that a resolver can access the database.

?

Reply:
sure